### PR TITLE
Prefer `scipy.spatial` `KDTree` over `cKDTree`

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

Before `scipy` v1.6.0 it used to be the case that `cKDTree` was more performant, but [since then `cKDTree` and `KDTree` are functionally identical](https://docs.scipy.org/doc/scipy/release/1.6.0-notes.html#scipy-spatial-improvements) and the [recommendation is to prefer `KDTree`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.cKDTree.html).

As far as I can tell this change should be invisible to normal users, other than very minor changes in some documentation.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
Since we require scipy 1.8 or later, 👍 but I'll let table maintainers review.

https://github.com/astropy/astropy/blob/56587db6e1cd7c9c5c8f8c7ee5a7e85a7b50ea1b/pyproject.toml#L68

**Review by neutrinoceros — APPROVED** [review]
LGTM, nice catch !

**Review by mhvk — APPROVED** [review]
Nice catch indeed!

**Pull request merged**
